example/lib/custom_marker.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:io';
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:mapbox_gl/mapbox_gl.dart';
+
+import 'main.dart';
+import 'page.dart';
+
+const RANDOM_MARKER_NUM = 100;
+
+class CustomMarker extends ExamplePage {
+  CustomMarker() : super(const Icon(Icons.place), 'Custom marker');
+
+  @override
+  Widget build(BuildContext context) {
+    return FullMap();
+  }
+}
+
+class FullMap extends StatefulWidget {
+  const FullMap();
+
+  @override
+  State createState() => FullMapState();
+}
+
+abstract class PositionChangeListener {
+  LatLng onPositionWillChange();
+  void onPositionChanged(Point point);
+}
+
+class FullMapState extends State<FullMap> {
+  final Random _rnd = new Random();
+
+  MapboxMapController _mapController;
+  List<Marker> _markers = [];
+  List<PositionChangeListener> _positionChangeListeners = [];
+
+  void _addPositionChangeListener(PositionChangeListener listener) {
+    _positionChangeListeners.add(listener);
+  }
+
+  void _onMapCreated(MapboxMapController controller) {
+    _mapController = controller;
+    controller.addListener(() {
+      if (controller.isCameraMoving) {
+        _updateMarkerPosition();
+      }
+    });
+  }
+
+  void _onStyleLoadedCallback() {
+    print('onStyleLoadedCallback');
+  }
+
+  void _onMapLongClickCallback(Point<double> point, LatLng coordinates) {
+    _addMarker(point, coordinates);
+  }
+
+  void _onCameraIdleCallback() {
+    _updateMarkerPosition();
+  }
+
+  void _updateMarkerPosition() {
+    var param = <LatLng>[];
```

Comment:
`final`, also should be renamed to reflect what the parameter actually represents. See comment in `method_channel_mapbox_gl.dart`

Reply:
Thanks, fixed